storage/fakestorage.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Rename this file to `something_test.go`, or move it to a test sub-package to be sure that we don't mix test-only things with production code?

Reply:
Fixed